compiler/optimizer/Inliner.cpp

```diff
@@ -1615,7 +1615,7 @@ static TR_YesNoMaybe gatherNodesToCheck(TR::Compilation *comp, TR_InlinerTracer
       for (uint16_t i = 0; i < currentNode->getNumChildren(); ++i)
          {
          // avoid exponential traversal
-         if (visitedNodes.ValueAt(currentNode->getGlobalIndex()))
+         if (visitedNodes.ValueAt(currentNode->getChild(i)->getGlobalIndex()))
```

Comment:
I assume this line's change in logic is intentional (?) It certainly seems like it makes more sense now with this commit.